core/src/main/java/org/elasticsearch/action/admin/indices/stats/CommonStatsFlags.java

```diff
@@ -54,6 +51,41 @@ public CommonStatsFlags(Flag... flags) {
         }
     }
 
+    public CommonStatsFlags(StreamInput in) throws IOException {
+        final long longFlags = in.readLong();
+        flags.clear();
+        for (Flag flag : Flag.values()) {
+            if ((longFlags & (1 << flag.ordinal())) != 0) {
+                flags.add(flag);
+            }
+        }
+        types = in.readStringArray();
+        groups = in.readStringArray();
+        fieldDataFields = in.readStringArray();
+        completionDataFields = in.readStringArray();
+        if (in.getVersion().onOrAfter(Version.V_5_0_0_alpha1)) {
```

Comment:
I am pushing a new commit, this tiny change doesn't probably deserve its own PR, I will make it a separate commit though.